mlir/include/mlir/Dialect/Polynomial/IR/Polynomial.td

```diff
@@ -149,7 +172,7 @@ def Polynomial_PolynomialType : Polynomial_Type<"Polynomial", "polynomial"> {
     A type for polynomials in a polynomial quotient ring.
   }];
   let parameters = (ins Polynomial_RingAttr:$ring);
-  let assemblyFormat = "`<` $ring `>`";
+  let assemblyFormat = "`<` qualified($ring) `>`";
```

Comment:
Is there a test change for this? What is the impact of leaving this out?

Reply:
Without this, a polynomial type will print like

```mlir
!polynomial.polynomial<<coefficientType = f32>>
```

When I would prefer the name of the attribute be present

```mlir
!polynomial.polynomial<#polynomial.ring<coefficientType = f32>>
```

The `<< ... >>` is what feels off to me. This is tested by `tools/mlir/test/Dialect/Polynomial/types.mlir`, and this specific line in this PR was added after the ring attribute custom parser was removed, because that custom parser printed the attribute name (cf. https://github.com/llvm/llvm-project/pull/91137/files#diff-93b00b22db2663bf648011bbdce7942738b0ba9a14a98f23404d6550eacca8dfL148)

Reply:
Isn't `#polynomial.ring` just pure noise here though? Seems a bit obnoxious to me to read the full expanded form.

Why not `!polynomial.polynomial<ring<coefficientType = f32>>` then?

Reply:
LG!